src/ios-deploy/ios-deploy.m

```diff
@@ -959,13 +963,27 @@ void write_lldb_prep_cmds(AMDeviceRef device, CFURLRef disk_app_url) {
 void
 server_callback (CFSocketRef s, CFSocketCallBackType callbackType, CFDataRef address, const void *data, void *info)
 {
-    if (CFDataGetLength (data) == 0) {
+    char buffer[0x4000];
```

Comment:
Is there a particular reason for this value?

From sniffing the packets, they don't seem to be very large. Plus looks like the while loop should handle larger packets.

Regardless of the size we should memset to null characters.

Reply:
It's nicer to handle everything in single read call, as I have some doubts in properly separating errors and no more data cases. Return value of 0 bytes mean both - ssl error and no more bytes. 

What I saw was mostly around 200-300 byte packets, but when large amount of stdout is printed, I did saw it climb to up to 3000 bytes.  Though I agree, 16 kb is a bit large, my intention was more in 4 kb.